File: src/main/java/br/com/blz/testjava/domain/factory/WarehousesFactory.java
package br.com.blz.testjava.domain.factory;

import java.util.ArrayList;
import java.util.List;

import br.com.blz.testjava.domain.model.Warehouse;

public class WarehousesFactory {
	
	private WarehousesFactory() {
    }

    public static List<Warehouse> zeroQuantityWarehouse() {
        List<Warehouse> warehouses = new ArrayList<>();
        warehouses.add(new Warehouse());

        return warehouses;
    }
}

File: src/main/kotlin/br/com/blz/testjava/application/factory/WarehousesFactory.kt
package br.com.blz.testjava.application.factory

import br.com.blz.testjava.application.model.WarehouseModel
import br.com.blz.testjava.domain.entity.Warehouse
import org.springframework.stereotype.Component

@Component
class WarehousesFactory: AbstractFactory<WarehouseModel, Warehouse>() {

    override fun create(model: WarehouseModel) = Warehouse(
        locality = model.locality,
        quantity = model.quantity,
        type = model.type
    )

}
